new device: ADVA FSP150CC and FSP3000R7 Series

czilian opened the pull request:
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/af672bc1-a308-4ed1-acea-e760144eb21d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/af672bc1-a308-4ed1-acea-e760144eb21d) completed: **No new issues**

czilian commented:
ADVA device support
-------------------
/images/os/adva.svg - created a squarish logo 
includes/definitions/adva.yaml - detects anything from ADVA
includes/polling/os/adva.inc.php - device specifics
tests/snmpsim/adva.snmprec - EGX as one type to detect
mibs/adva - ADVA specific MIBs but also some generic which are currently not in the MIBs root (maybe can move those from ADVA one level up later)

Sensor detection with two sections each for FSP150CC OS and FSP3000R7 OS:
discovery/sensors/pre-cache/adva.inc.php
discovery/sensors/current/adva.inc.php
discovery/sensors/dbm/adva.inc.php
discovery/sensors/temperature/adva.inc.php
discovery/sensors/voltage/adva.inc.php

laf commented on includes/definitions/adva.yaml:
```diff
@@ -0,0 +1,12 @@
+os: adva
```
I feel this is too generic. The os name should be based after the firmware/software the devices run, like IOS for Cisco switches.

If they don't have a name then at worst the os here should be something like adva-fsp.

czilian replied:
Thx for going through. These are two OS. Hence, splitting it into two yaml. All the sysObjectID checks in the other files will then also disappear. The OS names are unfortunately not catchy. Called it adva_fsp150 and adva_fsp3kr7 now.

laf commented on includes/discovery/sensors/current/adva.inc.php:
```diff
@@ -0,0 +1,126 @@
+// ***** Sensors for ADVA FSP150EG-X Chassis
+// ******************************************
+
+if (starts_with($device['sysObjectID'], 'enterprises.2544.1.12.1.1')) {
```
If this is to restrict the check to this OS then you can remove this if check as we only run sensors with the same filename as the os.

laf replied:
Looking at your other code it looks like it's to only perform the checks for specific devices.

Are you sure `$device['sysObjectID']` isn't a full numerical string like `.1.3.x.x.x.x`?

czilian replied:
gone.

laf commented on includes/discovery/sensors/current/adva.inc.php:
```diff
@@ -0,0 +1,126 @@
+
+if (starts_with($device['sysObjectID'], 'enterprises.2544.1.12.1.1')) {
+    // Define Sensors and Limits
+    $sensors = array
```
Please rename this variable, it's most likely going to conflict somewhere.

czilian replied:
gone.

laf commented on includes/discovery/sensors/current/adva.inc.php:
```diff
@@ -0,0 +1,126 @@
+                        'high_warn_limit' => 25,
+                        'high_limit'      => 30));
+    
+    foreach (array_keys($pre_cache['fsp150']) as $index1) {
```
Is this double foreach needed?

czilian replied:
Not for only one sensor, true. However, suggest to leave it still like this because then one can easily add another sensor into the $sensor array.

laf commented on includes/discovery/sensors/current/adva.inc.php:
```diff
@@ -0,0 +1,126 @@
+        foreach (array_keys($pre_cache['fsp3kr7_Card']) as $index) {
+        //AC PSU Limits
+            if ($pre_cache['fsp3kr7_Card'][$index]['eqptPhysInstValuePsuAmpere']) {
+                if ($pre_cache['fsp3kr7_Card'][$index]['eqptPhysInstValuePsuAmpere'] > 1000) {
```
Combine this if check with the one above.

czilian replied:
Sorry, this was unfinished and forgotten business. Wanted to distinguish between AC/DC PSUs to set the correct limits. Since I kicked out the hardcoded limits, this went away as well.

laf commented on includes/discovery/sensors/current/adva.inc.php:
```diff
@@ -0,0 +1,126 @@
+                    $high_limit = 260;
+                }
+        //DC PSU Limits
+                $low_limit = 35;
```
Doesn't this just reset what you defined above?

high/low limit should rarely be hard coded, we will calculate those.

czilian replied:
see comment before.

laf commented on includes/discovery/sensors/current/adva.inc.php:
```diff
@@ -0,0 +1,126 @@
+                    'current',
+                    $device,
+                    $oid,
+                    $index1,
```
If $index1 is a number, can it not be the same in your next discover_sensor() below? If so, I'd suggest using the OID name as the index like `slotCardUnitName` (but not that as it's not right).

czilian replied:
Will check. Less is more.

laf commented on includes/discovery/sensors/dbm/adva.inc.php:
```diff
@@ -0,0 +1,85 @@
+// other AidStrings to be inclued.
+        if ($entry['entityFacilityAidString'] and $entry['pmSnapshotCurrentInputPower']) {
+            $oidRX = '.1.3.6.1.4.1.2544.1.11.7.7.2.3.1.2.' . $index;
+            $limit_low                 = -20;
```
Same comment about limits here as well.

czilian replied:
done

laf commented on includes/discovery/sensors/dbm/adva.inc.php:
```diff
@@ -0,0 +1,85 @@
+                'dbm',
+                $device,
+                $oidRX,
+                $descr,
```
What is basically index here, needs to be uniqie so this should also have the $index appended to it.

czilian replied:
on this one will ping you in IRC. Also not happy with this. In fact ideally want to show Tx/Rx value in one mini-graph associated to the port. It's for a Layer1 device. Hence, no traffic but dBm readings. No clue how yet.

laf commented on includes/discovery/sensors/dbm/adva.inc.php:
```diff
@@ -0,0 +1,85 @@
+                'dbm',
+                $device,
+                $oidTX,
+                $descr,
```
Same comment as above.

laf commented on includes/discovery/sensors/dbm/adva.inc.php:
```diff
@@ -0,0 +1,85 @@
+
+        if ($entry['entityFacilityAidString'] and $entry['pmSnapshotCurrentOutputPower']) {
+            $oidTX = '.1.3.6.1.4.1.2544.1.11.7.7.2.3.1.1.' . $index;
+            $limit_low                 = -20;
```
Same comment about high / low limits.

czilian replied:
done

laf commented on includes/discovery/sensors/temperature/adva.inc.php:
```diff
@@ -0,0 +1,243 @@
+<?php
```
The same comments as per the other files apply here.

laf commented on includes/discovery/sensors/voltage/adva.inc.php:
```diff
@@ -0,0 +1,180 @@
+<?php
```
The same comments as per the other files apply here.

laf commented on includes/polling/os/adva.inc.php:
```diff
@@ -0,0 +1,34 @@
+// ***********  FSP150 Devices
+if (starts_with($device['sysObjectID'], 'enterprises.2544.1.12.1.1')) {
+    $version  = 'FSP150 SW V'.trim(snmp_get($device, "entPhysicalSoftwareRev.1", "-OQv", "ADVA-MIB"), '"');
+    $hardware = 'ADVA '.trim(snmp_get($device, "entPhysicalName.1", "-OQv", "ADVA-MIB"), '"')
```
Not sure we need to prepend ADVA here.

czilian replied:
No. Took it out.

laf commented on includes/polling/os/adva.inc.php:
```diff
@@ -0,0 +1,34 @@
+
+// ***********  FSP150 Devices
+if (starts_with($device['sysObjectID'], 'enterprises.2544.1.12.1.1')) {
+    $version  = 'FSP150 SW V'.trim(snmp_get($device, "entPhysicalSoftwareRev.1", "-OQv", "ADVA-MIB"), '"');
```
Not sure we need to prepend FSP150 SW V here.

czilian replied:
No. Took it out.

laf commented on includes/polling/os/adva.inc.php:
```diff
@@ -0,0 +1,34 @@
+    $hardware = 'ADVA '.trim(snmp_get($device, "entPhysicalName.1", "-OQv", "ADVA-MIB"), '"')
+                .' V'.trim(snmp_get($device, "entPhysicalHardwareRev.1", "-OQv", "ADVA-MIB"), '"');
+    $serial = trim(snmp_get($device, "entPhysicalSerialNum.1", "-OQv", "ADVA-MIB"), '"');
+    $features = ''; //search for PTP info in MIB
```
You don't need to set this if you aren't populating it.

czilian replied:
Unfinished business. Took it out for now. Want to pull license based features for which I still need to search for the OID

laf commented on includes/polling/os/adva.inc.php:
```diff
@@ -0,0 +1,34 @@
+
+// **********  FSP3000 R7 Devices
+if (starts_with($device['sysObjectID'], 'enterprises.2544.1.11.1.1')) {
+    $version  = 'FSP3000R7 SW V'.trim(snmp_get($device, "swVersionActiveApplSw.100737280", "-OQv", "ADVA-MIB"), '"');
```
Same comments as above.

laf commented on mibs/adva/8021ae.mib:
```diff
@@ -0,0 +1,1951 @@
+--
+-- *****************************************************************
+
+IEEE8021-SECY-MIB DEFINITIONS ::= BEGIN
```
This mib already exists: ``mibs/IEEE8021-SECY-MIB`.

I expect it's the same for a lot of the others, please remove the duplicates.

czilian replied:
done.

laf commented on tests/snmpsim/adva.snmprec:
```diff
@@ -0,0 +1,2 @@
+1.3.6.1.2.1.1.1.0|4|FSP150EG-X
```
Is this the full sysDescr value?

czilian replied:
For this one yes. But added sim-files for all types of ADVA boxes I have now.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/3d329f58-a30d-4507-9a14-7ef2f07c3cd9) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/3d329f58-a30d-4507-9a14-7ef2f07c3cd9) completed: **4 new issues**

czilian commented:
@laf 
OS for FSP150 and FSP3000R7 now in two seperate OS definitions
- split the files
- removed sysObjectId check for each
- removed hardcoded sensor limits
- removed more generic MIBs

laf commented on includes/definitions/adva_fsp150.yaml:
```diff
@@ -0,0 +1,13 @@
+os: adva_fsp150
+text: ADVA_FSP150CC
```
These are display texts so you should do them as human readable like Adva FSP150CC

czilian replied:
thx @laf. Remarks all understood and accepted. Only regarding the HW release I consider this as useful for the Ops people to immediately see which generation is in use. Will be back...

laf commented on html/images/os/tplink.svg:
```diff
@@ -1 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256"><g fill="#4accd5"><path d="M74.661 255.718c-3.842-.944-7.253-3.727-9.233-7.533l-1.097-2.11-.235-27.195-.235-27.194-27.47-.235-27.47-.234-2.165-1.272c-2.346-1.377-4.61-3.714-5.925-6.12C.067 182.43 0 180.382 0 158.396v-23.912h121.148v121.438l-22.539.072c-12.396.039-23.173-.085-23.948-.275z"/><path d="M140.873 189.514c-.136-.353-.192-12.4 ...
\ No newline at end of file
+<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 210.66 251.52"><defs><style>.cls-1{fill:#4accd5;}.cls-2{isolation:isolate;}</style></defs><title>tplink</title><path class="cls-1" d="M63.63,207.44a11.67,11.67,0,0,1-7.39-6l-.88-1.69L55.17,178,55,156.19,33,156l-22-.19-1.73-1a13.52,13.52,0,0,1-4.74-4.9c-.61-1.12-.67-2.76-.67-20.36V110.36h97v97.25l-18,.06c-9.93,0- ...
```
Can you revert this file as it shouldn't be part of this PR.

laf commented on html/images/os/adva.svg:
```diff
@@ -0,0 +1 @@
+<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 328.07 318.86"><defs><style>.cls-1{fill:none;}.cls-2{fill:#e80707;}.cls-3{fill:#3950a3;stroke:#305e84;stroke-miterlimit:10;}.cls-4{fill:#004588;}.cls-5{fill:#28e87e;}</style></defs><title>adva</title><line class="cls-1" x1="287.28" y1="281.75" x2="322.32" y2="281.75"/><line class="cls-1" x1="289.03" y1="288.25"  ...
```
OS logos should be Y x Y (square), this is slightly different in width. Can you stretch it out?

laf commented on includes/discovery/sensors/current/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,61 @@
+            if ($pre_cache['fsp150'][$index1][$sensor_name]) {
+                $descr       = $pre_cache['fsp150'][$index1]['slotCardUnitName']." [#".$pre_cache['fsp150'][$index1]['slotIndex']."]";
+                $current     = $pre_cache['fsp150'][$index1][$entry];
+                $sensorType  = 'advafsp150';
```
This should match the OS name so: adva_fsp150

laf commented on includes/discovery/sensors/current/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,61 @@
+                    'current',
+                    $device,
+                    $oid,
+                    $index1,
```
You should replace this with `"psuOutputCurrent.$index1",`

laf commented on includes/discovery/sensors/current/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+            $slotnum    = $index;
+            $descr      = strtoupper($pre_cache['fsp3kr7_Card'][$index]['entityEqptAidString'])." Input";
+            $current    = $pre_cachen['fsp3kr7_Card'][$index]['eqptPhysInstValuePsuAmpere'];
+            $sensorType = 'advafsp3kr7';
```
This should be the os name: adva_fsp3kr7

laf commented on includes/discovery/sensors/current/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+                'current',
+                $device,
+                $oid,
+                $index,
```
You should replace this with `"eqptPhysInstValuePsuAmpere.$index",`

laf commented on includes/discovery/sensors/current/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,61 @@
+                    $descr,
+                    $divisor,
+                    $multiplier,
+                    $low_limit,
```
As you haven't defined the limits please set them as `null`

laf commented on includes/discovery/sensors/current/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+                $descr,
+                $divisor,
+                $multiplier,
+                $low_limit,
```
As you haven't defined the limits please set them as `null`

laf commented on includes/discovery/sensors/dbm/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,77 @@
+            $device,
+            $oidRX,
+            $descr,
+            'advafsp3kr7',
```
Should be the os name

laf commented on includes/discovery/sensors/dbm/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,77 @@
+            $descr,
+            $divisor,
+            $multiplier,
+            $limit_low,
```
As you haven't defined the limits please set them as `null`

laf commented on includes/discovery/sensors/dbm/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,77 @@
+            'dbm',
+            $device,
+            $oidRX,
+            $descr,
```
This should be `"pmSnapshotCurrentInputPower.$index",`

laf commented on includes/discovery/sensors/dbm/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,77 @@
+
+    if ($entry['entityFacilityAidString'] and $entry['pmSnapshotCurrentOutputPower']) {
+        $oidTX = '.1.3.6.1.4.1.2544.1.11.7.7.2.3.1.1.' . $index;
+        $limit_low                 = -20;
```
If you don't need to you should avoid setting limits statically. It is fine, just be sure they are correct.

laf commented on includes/discovery/sensors/dbm/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,77 @@
+            'dbm',
+            $device,
+            $oidTX,
+            $descr,
```
This should be `"pmSnapshotCurrentOutputPower.$index",`

laf commented on includes/discovery/sensors/dbm/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,77 @@
+            $device,
+            $oidTX,
+            $descr,
+            'advafsp3kr7',
```
Should be the os name

laf commented on includes/discovery/sensors/pre-cache/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,20 @@
+ **/
+
+// FSP150CC Series
+$pre_cache['fsp150'] = snmpwalk_cache_multi_oid($device, 'cmEntityObjects', array(), 'CM-ENTITY-MIB', '/opt/librenms/mibs/adva', '-OQUbs');
```
You don't need to set the mib dir, just set to `null`

laf commented on includes/discovery/sensors/pre-cache/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,39 @@
+ **/
+
+// FSP3000 R7 Series
+$pre_cache['fsp3kr7'] = snmpwalk_cache_multi_oid($device, 'pmSnapshotCurrentEntry', $pre_cache['fsp3kr7'], 'ADVA-FSPR7-MIB', '/opt/librenms/mibs/adva', '-OQUbs');
```
You shouldn't set the mib dir for any of these, just set `null`

laf commented on includes/discovery/sensors/temperature/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,103 @@
+            if ($pre_cache['fsp150'][$index1][$sensor_name]) {
+                $descr       = $pre_cache['fsp150'][$index1]['slotCardUnitName']." [#".$pre_cache['fsp150'][$index1]['slotIndex']."]";
+                $current     = $pre_cache['fsp150'][$index1][$entry];
+                $sensorType  = 'advafsp150';
```
Should be the OS name.

laf commented on includes/discovery/sensors/temperature/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,103 @@
+                    'temperature',
+                    $device,
+                    $oid,
+                    $index1,
```
Should be `"$sensor_name.$index1",`

laf commented on includes/discovery/sensors/temperature/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,103 @@
+                    $descr,
+                    $divisor,
+                    $multiplier,
+                    $low_limit,
```
Limits should be null if you haven't set them.

laf commented on includes/discovery/sensors/temperature/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,103 @@
+// ***** Temperature Sensors for ADVA FSP150CC Series
+// *************************************************************
+
+    $sensors = array
```
Please rename this to save conflicts.

laf commented on includes/discovery/sensors/temperature/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+            $slotnum    = $index;
+            $descr      = $pre_cache['fsp3kr7_Card'][$index]['entityEqptAidString'];
+            $current    = $pre_cache['fsp3kr7_Card'][$index]['eqptPhysInstValueTemp'];
+            $sensorType = 'advafsp3kr7';
```
Should be the os name.

laf commented on includes/discovery/sensors/temperature/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+                'temperature',
+                $device,
+                $oid,
+                $index,
```
Should be `"eqptPhysInstValueTemp.$index",`

laf commented on includes/discovery/sensors/temperature/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+                $descr,
+                $divisor,
+                $multiplier,
+                $low_limit,
```
limits should be set to `null` if you don't have them defined.

laf commented on includes/discovery/sensors/voltage/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,80 @@
+// ******************************************
+
+    // Define Sensors
+    $sensors = array
```
Please rename this array name.

laf commented on includes/discovery/sensors/voltage/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,80 @@
+            if ($pre_cache['fsp150'][$index1][$sensor_name]) {
+                $descr       = $pre_cache['fsp150'][$index1]['slotCardUnitName']." [#".$pre_cache['fsp150'][$index1]['slotIndex']."]";
+                $current     = $pre_cache['fsp150'][$index1][$entry];
+                $sensorType  = 'advafsp150';
```
This should be the os name.

laf commented on includes/discovery/sensors/voltage/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,80 @@
+                    'voltage',
+                    $device,
+                    $oid,
+                    $index1,
```
Should be `"$sensor_name.$index1",`

laf commented on includes/discovery/sensors/voltage/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,80 @@
+                    $descr,
+                    $divisor,
+                    $multiplier,
+                    $low_limit,
```
Limits should be set to `null`

laf commented on includes/discovery/sensors/voltage/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+            $slotnum    = $index;
+            $descr      = strtoupper($pre_cache['fsp3kr7_Card'][$index]['entityEqptAidString'])." Input";
+            $current    = $pre_cache['fsp3kr7_Card'][$index]['eqptPhysInstValuePsuVoltInp'];
+            $sensorType = 'advafsp3kr7';
```
Should be the os name.

laf commented on includes/discovery/sensors/voltage/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+                'voltage',
+                $device,
+                $oid,
+                $index,
```
Should be `"eqptPhysInstValuePsuVoltInp.$index",`

laf commented on includes/discovery/sensors/voltage/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,55 @@
+                $descr,
+                $divisor,
+                $multiplier,
+                $low_limit,
```
Should be set to `null` if you haven't defined them.

laf commented on includes/polling/os/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,22 @@
+// ***********  FSP150 Devices
+$version  = trim(snmp_get($device, "entPhysicalSoftwareRev.1", "-OQv", "ADVA-MIB"), '"');
+$hardware = trim(snmp_get($device, "entPhysicalName.1", "-OQv", "ADVA-MIB"), '"')
+            .' V'.trim(snmp_get($device, "entPhysicalHardwareRev.1", "-OQv", "ADVA-MIB"), '"');
```
I'm not sure we should store the hardware rev

laf commented on includes/polling/os/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,22 @@
+// **********  FSP3000 R7 Devices
+$version  = trim(snmp_get($device, "swVersionActiveApplSw.100737280", "-OQv", "ADVA-MIB"), '"');
+$hardware = trim(snmp_get($device, "inventoryUnitName.33619968", "-OQv", "ADVA-MIB"), '"')
+            .' V'.trim(snmp_get($device, "inventoryHardwareRev.33619968", "-OQv", "ADVA-MIB"), '"');
```
I'm not sure we should store the hardware rev

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/8a4329e8-16a7-4cae-b0db-b74032e17427) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/8a4329e8-16a7-4cae-b0db-b74032e17427) completed: **No new issues**

czilian commented:
@laf Greetings. The variables related stuff should be all done. For the RRD filename I rather want to use the more meaningful definition as you can see here.
![2017-05-23 1](https://cloud.githubusercontent.com/assets/19749010/26353185/b8c62640-3ff1-11e7-9cc4-f818198a095e.png)
The adva mib was searching for 2 LLDP mibs. Hence, put the one into the ADVA mib folder.
![2017-05-23](https://cloud.githubusercontent.com/assets/19749010/26353231/f910dcea-3ff1-11e7-9811-245a87a71281.png)
Regarding the OS logos: 
1.) the "improved" tplink.svg sneaked into this branch. Problem is I don't know how to get back your original one. Tried checkout -- filename but this just deleted mine. Tried now by copying your original version into this branch.
2.) ADVA logo. Not sure why the logo is not 100% because it is an export from an AI art-board 320x320.

czilian commented:
Anything you still want me to change?

laf commented on includes/discovery/sensors/current/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,64 @@
+            $sensor_name = $entry['sensor_name'];
+            if ($pre_cache['adva_fsp150'][$index][$sensor_name]) {
+                $oid          = $entry['sensor_oid'].".".$index;
+                $rrd_filename = $pre_cache['adva_fsp150'][$index]['slotCardUnitName']."-".$pre_cache['adva_fsp150'][$index]['slotIndex'];
```
I really think this is over complicating things. `$rrd_filename` should be named `$index` anyway as that's what the variable is but the setting of how it is is overkill and should be simplified to `psuOutputCurrent.$index`

This can be looked up in the db if more info is needed to work out what that index references.

laf commented on includes/discovery/sensors/current/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,64 @@
+                $oid          = $entry['sensor_oid'].".".$index;
+                $rrd_filename = $pre_cache['adva_fsp150'][$index]['slotCardUnitName']."-".$pre_cache['adva_fsp150'][$index]['slotIndex'];
+                $descr        = $pre_cache['adva_fsp150'][$index]['slotCardUnitName']." [#".$pre_cache['adva_fsp150'][$index]['slotIndex']."]";
+                $current      = $pre_cache['adva_fsp150'][$index][$entry];
```
This needs to be divided by `$divisor` otherwise high/low values will be wrong.

laf commented on includes/discovery/sensors/current/adva_fsp150.inc.php:
```diff
@@ -0,0 +1,64 @@
+                    $device,
+                    $oid,
+                    $rrd_filename,
+                    $sensorType,
```
You should just put `'adva_fsp150`' here

laf commented on includes/discovery/sensors/current/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,53 @@
+                $device,
+                $oid,
+                $rrd_filename,
+                $sensorType,
```
You should just put `'adva_fsp3kr7'` here.

laf commented on includes/discovery/sensors/current/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,53 @@
+        if ($pre_cache['adva_fsp3kr7_Card'][$index]['eqptPhysInstValuePsuAmpere']) {
+            $oid          = '.1.3.6.1.4.1.2544.1.11.11.1.2.1.1.1.6.'.$index;
+            $descr        = strtoupper($pre_cache['adva_fsp3kr7_Card'][$index]['entityEqptAidString'])." Input";
+            $rrd_filename = $descr;
```
Same as the previous file, rrd_filename should be index.

laf commented on includes/discovery/sensors/current/adva_fsp3kr7.inc.php:
```diff
@@ -0,0 +1,53 @@
+            $oid          = '.1.3.6.1.4.1.2544.1.11.11.1.2.1.1.1.6.'.$index;
+            $descr        = strtoupper($pre_cache['adva_fsp3kr7_Card'][$index]['entityEqptAidString'])." Input";
+            $rrd_filename = $descr;
+            $current      = $pre_cache['adva_fsp3kr7_Card'][$index]['eqptPhysInstValuePsuAmpere'];
```
Needs dividing by `$divisor`

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/cb2471f6-1fb3-41a8-9b95-5504453d87af) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/cb2471f6-1fb3-41a8-9b95-5504453d87af) completed: **No new issues**

laf commented:
![image](https://cloud.githubusercontent.com/assets/3941142/26742875/7a6fed52-47d7-11e7-8aea-1814d090503a.png)
